### PR TITLE
fix: sync massive number repos catalog failed

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**

- Can't sync terraform modules due to massive number of repos

**Solution:**

- provider catalog regexp pattern to filter repos with name

**Related Issue:**
#2150

### CODE REVIEW

File: pkg/catalog/catalog.go [hunk 1]
```diff
@@ -64,15 +65,24 @@ func getRepos(ctx context.Context, c *model.Catalog, ua string, skipTLSVerify bo
 		return nil, err
 	}
 
-	list := make([]*vcs.Repository, len(repos))
+	var nameReg *regexp.Regexp
+	if c.FilterPattern != "" {
+		nameReg = regexp.MustCompile(c.FilterPattern)
```
Comment (gitlawr) [inline pkg/catalog/catalog.go:70]: Avoid using MustCompile, it causes panic

Comment (alexcodelf) [reply pkg/catalog/catalog.go]: updated

File: pkg/apis/catalog/basic_view.go [hunk 2]
```diff
@@ -40,6 +41,10 @@ func (r *CreateRequest) Validate() error {
 		return fmt.Errorf("unsupported catalog type %q", r.Type)
 	}
 
+	if _, err = regexp.Compile(r.FilterPattern); err != nil {
+		return err
```
Comment (gitlawr) [inline pkg/apis/catalog/basic_view.go:45]: How does the error look like? Should we make it more friendly

Comment (alexcodelf) [reply pkg/apis/catalog/basic_view.go:45]: image
If regexp is not vaild, this is how it looks like , shall we warp this error?

Comment (gitlawr) [reply pkg/apis/catalog/basic_view.go:45]: fmt.Errorf("invalid filter pattern: %w", err)

Comment (alexcodelf) [reply pkg/apis/catalog/basic_view.go]: Then the error will be:
```json
{
  "message": "invalid filter pattern: error parsing regexp: missing closing ]: `[email\\]@example.com`",
  "status": 400,
  "statusText": "Bad Request"
}
```

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**